test/SlowTests/Issues/RavenDB_17046.cs

```diff
@@ -0,0 +1,105 @@
+﻿using System.Collections.Generic;
+using FastTests;
+using Orders;
+using Raven.Client.Documents;
+using Xunit;
+using Xunit.Abstractions;
+using Raven.Client.Documents.Session;
+
+namespace SlowTests.Issues
+{
+    public class RavenDB_17046 : RavenTestBase
+    {
+        public RavenDB_17046(ITestOutputHelper output) : base(output)
+        {
+        }
+
+        [Theory]
+        [InlineData("c")]
+        [InlineData("co")]
+        public void CheckIfPagingWorksForUnboundedSet(string startsWithParam)
+        {
+            var result = Act<Company>("Companies", startsWithParam);
+
+            result.wholeCollection.Sort(CompaniesSortById);
+            result.wholeCollectionByPaging.Sort(CompaniesSortById);
+            Assert.Equal(result.wholeCollection.Count, result.wholeCollectionByPaging.Count);
+            Assert.Equal(result.wholeCollection, result.wholeCollectionByPaging);
+        }
+
+        [Fact]
+        public void CheckIfPagingWorksForUnboundedSetOnAllDocs()
+        {
+            var result = Act<dynamic>("@all_docs", "c");
+            Assert.Equal(result.wholeCollection.Count,result.wholeCollectionByPaging.Count);
+        }
+
+        private (List<T> wholeCollection, List<T> wholeCollectionByPaging) Act<T>(string collection, string startsWithParam)
+        {
+            string fromStatement = $"from {collection}";
+            string fullStatement = fromStatement + $" where startsWith(id(),'{startsWithParam}')";
+            List<T> wholeCollection;
+            List<T> wholeCollectionByPaging;
+
+            using (var store = GetDocumentStore())
+            {
+                PrepareDatabaseForTest(store);
+                using (var session = store.OpenSession())
+                {
+                    wholeCollection = session.Advanced.RawQuery<T>(fromStatement).ToList();
```

Comment:
As we discussed  - for assertion purposes it's better to check it against the values generated inside the test instead of relaying on the results of another query (even if it's raw query).

Reply:
Done